### ISSUE TITLE
Typings's broken on latest Typescript (2.5.3)

### ISSUE THREAD

**Issue by dashmug** [date-fns/date-fns#572]
I am not sure why but something in v2.5.3 has broken compilation of my project.

`Could not find a declaration file for module 'date-fns/format'. '/path/to/my/project/node_modules/date-fns/format/index.js' implicitly has an 'any' type.`

I'm using `date-fns` like this

```js
import * as format from 'date-fns/format'
```

For now, I've reverted it back to Typescript v2.5.2 as it is working with no problems.

**Comment by egorshulga** [comment]
I confirm the issue.
Date-fns version is 2.0.0-alpha.6.

**Comment by dashmug** [comment]
I can also confirm that it works if we import it using

```
import { format } from 'date-fns'
```

**Comment by luisbelloch** [comment]
TS 2.5.3 changes [here](https://github.com/Microsoft/TypeScript/milestone/55?closed=1). I cannot tell the change that broke the typings.

Works for me using Typescript 2.5.2 and date-fns 1.28.5

**Comment by kossnocorp** [comment]
If someone could investigate the reason and help me to fix it, I will much appreciate it! ❤️  I'm not a TS user so it's hard to me dig into it.

**Comment by kossnocorp** [comment]
@dashmug which version you have btw?

**Comment by dashmug** [comment]
@kossnocorp 
I'm using `date-fns v1.28.5` and `typescript v2.5.3`.

This looks like a bug with Typescript. Here's what I found so far...

If I comment out [lines 1128 to 1131 of typings.d.ts](https://github.com/date-fns/date-fns/blob/v1/typings.d.ts#L1128-L1131) 

```ts
// declare module 'date-fns/format' {
//     import { format } from 'date-fns'
//     export = format
// }

```

and move it into another file in my project (say `typings/date-fns.d.ts`), 

```ts
declare module 'date-fns/format' {
    import { format } from 'date-fns'
    export = format
}
```

it just works as usual.

Could it be that Typescript does not allow multiple `declare module` statements in a single file anymore? I don't see any commit for that.

**Comment by plantain-00** [comment]
According to feedback from typescript maintainers: 
>A solution that should work in both old and new versions of TypeScript would be to create a file date-fns/format/index.d.ts like is done with date-fns/format/index.js.flow, instead of adding many declare module in date-fns/typings.d.ts. Then there would be no need for adding typings to package.json since the definitions would be where we expect them to be anyway.

And I tested it locally, it should work by creating `index.d.ts` file in `node_modules/date-fns/format/` with following code:

```ts
declare module 'date-fns/format' {
    import { format } from 'date-fns'
    export = format
}
```

**Comment by dashmug** [comment]
@kossnocorp 
Here's my attempt to restructure the typings: #573 

I'm only basing it on the `v1` branch.

Feedback is welcome.

**Comment by dashmug** [comment]
@kossnocorp I'm sorry I wasn't able to find time to continue this. I'll close this so someone else can continue.

**Comment by kossnocorp** [comment]
@dashmug hey, I've just merged https://github.com/date-fns/date-fns/pull/573. That was a mistake? I'm a bit confused because you've closed the issue but not the PR. Can you tell me more about the current state and if I should rollback the PR.

**Comment by dashmug** [comment]
@kossnocorp My bad. I was supposed to close the PR. It wasn't working and I currently don't have the time (in the near future) to continue it.

Please rollback the merge. My apologies for the confusion.

**Comment by kossnocorp** [comment]
Ok, no worries! Thank you for your attempt to help!

**Comment by kossnocorp** [comment]
If anyone wants to chime in and help to solve the issue, you're very welcome!

**Comment by Hotell** [comment]
temporary fix is easy enough:

just use type path mapping to your custom override of `format`

```json
{
  "baseUrl": "./",
  "paths": {
      "date-fns/format": [
        "manual_typings/date-fns/format"
      ]
    }
}
```

```
  /manual_typings
  | - /date-fns
  |    | - /format
  |    |    |- index.d.ts
```

```ts
// manual_typings/date-fns/format/index.d.ts
declare module 'date-fns/format' {
  import {format} from 'date-fns'
  export = format
}
```

**Comment by plantain-00** [comment]
Is there someone that can review #598 and #599 ? This issue need to be fixed.

**Comment by corydeppen** [comment]
Looks like both have been merged. Can you give an idea when a new release will be available?

**Comment by havenchyk** [comment]
@kossnocorp @leshakoss we need fixed typings 😄 could you please make a release?

**Comment by howtimeflies-io** [comment]
Hey guys, a quick fix is using the `require` statement:

```const fromat = require('date-fns/format/index')```

**Comment by jsanta** [comment]
I'm having this same issue using v1.29 
Used @howtimeflies-io workaround with a couple of fixs:
- After my imports I added ```declare var require: any;``` because _require_ wasn't found.
- Inside my class I placed teh workaround line ```format = require('date-fns/format/index');``` without the const.
- Afterwards I was able to use _format_ like (eg) 
```let today: string = this.format(new Date(), 'YYYY-MM-DD');```

I know it's not the best solution, but at least got this to work. Hope it helps anyone facing the same problem.
Best regards.
